internal/pkg/api/cluster.go

```diff
@@ -126,6 +126,26 @@ func WithControlPlaneLabel(key string, val string) ClusterFiller {
 	}
 }
 
+// WithControlPlaneAPIServerExtraArgs adds the APIServerExtraArgs to the cluster spec.
+func WithControlPlaneAPIServerExtraArgs() ClusterFiller {
```

Comment:
I didn't notice this before. Using this function name makes me think this should be far more flexible than only supporting JWKS args. We can either figure out a way to make it more flexible, or rename it to be specific to handling JWKS.

Reply:
You are right about that. We updated the design to allow configuring any of the flags in the map through a feature flag